### PR TITLE
refactor: logits as attributes of FeatureExtractor

### PR DESCRIPTION
**Pull request by paulnovello**
* Refactor answering #73 
* Also add missing tests to check the output of .score().

### CODE REVIEW

**Review by y-prudent — COMMENTED** [review]
LGTM

**Review by y-prudent — APPROVED** [review]
